src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts

```diff
@@ -606,6 +606,11 @@ const terminalConfiguration: IConfigurationNode = {
 			],
 			default: 'both'
 		},
+		[TerminalSettingId.ShellIntegrationEnvironmentReporting]: {
+			markdownDescription: localize('terminal.integrated.shellIntegration.environmentReporting', "Controls whether to use terminal environment API to report environment state to VS Code"),
+			type: 'boolean',
+			default: true
```

Comment:
With this we can short circuit the stable check and do it in TS instead using this:

https://github.com/microsoft/vscode/blob/2ffdbd3cd9045de21143fc28d7e7f0223d1b8db7/src/vs/workbench/contrib/chat/browser/chat.contribution.ts#L264

When we do that we can remove the stable condition from the shell scripts, any simplification we can do there is good.